src/amo/components/CollectionSort/index.js

```diff
@@ -0,0 +1,145 @@
+import {
+  COLLECTION_SORT_DATE_ADDED_ASCENDING,
+  COLLECTION_SORT_DATE_ADDED_DESCENDING,
+  COLLECTION_SORT_NAME,
+  COLLECTION_SORT_POPULARITY,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import Card from 'ui/components/Card';
+import Select from 'ui/components/Select';
+import type {
+  CollectionFilters,
+  CollectionType,
+} from 'amo/reducers/collections';
+import type { AppState } from 'amo/store';
+import type { I18nType } from 'core/types/i18n';
+import type { ReactRouterType } from 'core/types/router';
+
+import './styles.scss';
+
+export type Props = {|
+  collection: CollectionType | null,
+  editing: boolean,
+  filters: CollectionFilters,
+|};
+
+type InternalProps = {|
+  ...Props,
+  clientApp: string,
+  i18n: I18nType,
+  lang: string,
+  router: ReactRouterType,
+|};
+
+export class CollectionSortBase extends React.Component<InternalProps> {
+  onSortSelect = (event: SyntheticEvent<HTMLSelectElement>) => {
+    const {
+      collection,
+      clientApp,
+      editing,
+      filters,
+      lang,
+      router,
+    } = this.props;
+
+    invariant(collection, 'A collection is required.');
+
+    const collectionSort = event.currentTarget.value;
+    const newFilters = {
+      ...filters,
+      collectionSort,
+    };
+
+    const pathname = `/${lang}/${clientApp}${
+      editing
+        ? collectionEditUrl({ collection })
+        : collectionUrl({ collection })
+    }`;
+    router.push({
+      pathname,
+      query: convertFiltersToQueryParams(newFilters),
+    });
+  };
+
+  sortOptions() {
+    const { i18n } = this.props;
+
+    return [
+      {
+        label: i18n.gettext('Newest first'),
+        value: COLLECTION_SORT_DATE_ADDED_DESCENDING,
+      },
+      {
+        label: i18n.gettext('Oldest first'),
+        value: COLLECTION_SORT_DATE_ADDED_ASCENDING,
+      },
+      {
+        label: i18n.gettext('Name'),
+        value: COLLECTION_SORT_NAME,
+      },
+      {
+        label: i18n.gettext('Popularity'),
+        value: COLLECTION_SORT_POPULARITY,
+      },
+    ];
+  }
+
+  render() {
+    const { collection, filters, i18n } = this.props;
+
+    return collection ? (
+      <Card className="CollectionSort">
+        <form>
+          <label
+            className="CollectionSort-label"
+            htmlFor="CollectionSort-select"
+          >
+            {i18n.gettext('Sort add-ons by')}
+          </label>
+          <Select
+            className="CollectionSort-select"
+            defaultValue={filters.collectionSort}
+            id="CollectionSort-select"
+            name="sort"
+            onChange={this.onSortSelect}
+          >
+            {this.sortOptions().map((option) => {
+              return (
+                <option key={option.value} value={option.value}>
+                  {option.label}
+                </option>
+              );
+            })}
+          </Select>
+        </form>
+      </Card>
+    ) : null;
```

Comment:
I think it would be better to render this form even without a collection (i.e. when loading). It should render fine -- it would just throw an error if you clicked on a sort option before the collection loaded (which would be hard to do in real life). The old component worked like this.

It's a good habit to try and render as much as possible while data is loading. This makes the page less jumpy when the final content loads.